src/drivers/sdr/gnuradio.c

```diff
@@ -70,40 +75,48 @@ uint16_t crc16(uint8_t * pData, int length)
 
 //format and send 128B frame to gnuradio via zmq
 int csp_gnuradio_tx(int fd, const void * data, size_t len){
-
-    if(len != PACKET_LEN){
-        while(1); //improve error handling
+    
+    if((int)len != PACKET_LEN){
+        exit(); //todo support len != 128 one day, maybe
```

Comment:
Make this call exit(1) instead. This tells the calling program (shell for example) that this did not exit successfully.

Also, a print saying why it will exit might be a good idea